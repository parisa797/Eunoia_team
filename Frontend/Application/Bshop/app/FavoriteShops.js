import React, { useState, useEffect } from "react";
import {
  View,
  Text,
  Image,
  StyleSheet,
  ImageBackground,
  TouchableOpacity,
  TouchableNativeFeedback,
  Platform,
  FlatList,
  Pressable,
  ScrollView,
} from "react-native";
import Icon from "react-native-vector-icons/FontAwesome";
import Ionicon from "react-native-vector-icons/Ionicons";
import StarRating from "react-native-star-rating";
import { useIsFocused } from "@react-navigation/native";
import * as SecureStore from "expo-secure-store";
import Shop from "./shop";

const FavoriteShops = ({ navigation }) => {
  const [shops, setShops] = useState();
  const isFocused = useIsFocused();

  useEffect(() => {
    const getLikedShops = async () => {
      console.log("yes in shops!");
      var myHeaders = new Headers();
      let t = await SecureStore.getItemAsync("token");
      var authorization = "Token " + t;
      myHeaders.append("Authorization", authorization);

      var requestOptions = {
        method: "GET",
        headers: myHeaders,
        redirect: "follow",
      };
      fetch(
        "http://eunoia-bshop.ir:8000/users/profile/likedshops",
        requestOptions
      )
        .then((response) => response.json())
        .then((result) => {
          setShops(result);
          console.log(result);
        })
        .catch((error) => console.log("error", error));
    };
    getLikedShops();
  }, [isFocused]);

  return (
    <ScrollView nestedScrollEnabled={true} style={styles.container}>
      {shops && (
        <FlatList
          // testID={"items-list" + props.index}
          nestedScrollEnabled={true}
          style={{ marginTop: -30 }}
          data={shops}
          keyExtractor={(item) => item.id.toString()}
          renderItem={(itemData) => {
            // console.log("item is", itemData.item);
            return (
              <Shop
                title={itemData.item.title}
                address={itemData.item.address}
                image={itemData.item.logo}
                rate_value={itemData.item.rate_value}
                online={itemData.item.online}
                phone={itemData.item.phone}
                index={itemData.item.id}
                onSelect={() => {
                  navigation.navigate("ShopDetail", itemData.item);
                }}
              ></Shop>
            );
          }}
        />
      )}

      {/* </ImageBackground> */}
    </ScrollView>
  );
};

const styles = StyleSheet.create({
  SignUpBtn: {
    width: 70,
    borderRadius: 100,
    height: 40,
    alignItems: "center",
    justifyContent: "center",
    marginTop: -60,
    marginBottom: 20,
    backgroundColor: "#b31414",
    marginLeft: 10,
  },
  SignUpText: {
    color: "white",
  },
  // naghsheBtn: {
  //   width: 70,
  //   borderRadius: 100,
  //   height: 40,
  //   alignItems: "center",
  //   justifyContent: "center",
  //   marginTop: -60,
  //   marginBottom: 20,
  //   backgroundColor: "#b31414",
  //   marginLeft: 10,
  // },
  // naghsheText: {
  //   color: "white",
  // },
  icon: {
    justifyContent: "space-between",
    flexDirection: "row-reverse",
    alignItems: "baseline",
  },
  hozuri: { fontSize: 20, color: "red" },
  online: { fontSize: 20, color: "green" },
  shop: {
    borderRadius: 10,
    marginTop: 1,
    borderRadius: 10,
    shadowColor: "black",
    shadowOpacity: 0.26,
    shadowOffset: { width: 0, height: 2 },
    elevation: 5,
    backgroundColor: "#f1f1f2",
    height: 300,
    width: 370,
    marginLeft: 18,
  },
  rows: {
    borderRadius: 10,
    marginTop: 15,
    borderRadius: 10,
    shadowColor: "black",
    shadowOpacity: 0.26,
    shadowOffset: { width: 0, height: 2 },
    elevation: 5,
    backgroundColor: "#b31414",
    height: 40,
    width: 370,
    marginLeft: 18,
    fontWeight: "bold",
    fontSize: 25,
  },
  rowstext: {
    color: "white",
    fontSize: 20,
    marginTop: 5,
    textAlign: "center",
  },
  imagebackk: {
    width: 400,
    height: 700,
    marginTop: -20,
    marginLeft: -7,
  },
  items: {
    backgroundColor: "white",
    height: 210,
    width: 165,
    borderRadius: 10,
    marginTop: 15,
    borderRadius: 10,
    shadowColor: "black",
    shadowOpacity: 0.26,
    shadowOffset: { width: 1, height: 3 },
    elevation: 5,
    backgroundColor: "#f1f1f2",
    marginLeft: 12,
  },
  imageContainer: {
    backgroundColor: "white",
    // objectFit: ""
    width: "39%",
    height: "40%",
    // alignItems: "center",
    // justifyContent: "center",
    // alignContent: "center",
    shadowColor: "black",
    shadowOpacity: 0.26,
    shadowOffset: { width: 0, height: 2 },
    elevation: 5,
    // borderTopLeftRadius: 10,
    // borderTopRightRadius: 10,
    borderRadius: 10,
    marginLeft: 110,
    marginTop: 10,

    // overflow: "hidden",
  },
  image: {
    // backgroundColor: "#f1f1f2",
    width: "80%",
    height: "80%",
    marginTop: "12%",
    resizeMode: "cover",
    alignItems: "center",
    justifyContent: "center",
    // paddingBottom: "1",
    marginLeft: 12,
  },
  details: {
    alignItems: "center",
    height: "38%",
    padding: 10,
  },
  title: {
    // fontFamily: "open-sans-bold",
    fontSize: 20,
    marginVertical: 5,
    // paddingTop: 3,
    marginLeft: "-1%",
    marginTop: "1%",
  },
  online_icon: {
    justifyContent: "space-between",
    flexDirection: "row-reverse",
    backgroundColor: "transparent",
    alignItems: "baseline",
  },
  address: {
    // fontFamily: "open-sans",
    fontSize: 20,
    color: "black",
  },
  mantaghee: {
    fontSize: 21,
    color: "black",
    marginLeft: 20,
  },
  container: {
    flex: 1,
    // backgroundColor: "#fff",
    // alignItems: "center",
    // justifyContent: "center",
    marginTop: "5%",
    // width: "50%",
    // height: "50%",
    marginBottom: "5%",
  },
});
export default FavoriteShops;
