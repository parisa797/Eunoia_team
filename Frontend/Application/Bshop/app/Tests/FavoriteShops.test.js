import React from "react";
import {
  render,
  fireEvent,
  cleanup,
  waitFor,
  act,
} from "react-native-testing-library";
import renderer from "react-test-renderer";
import { useIsFocused } from "@react-navigation/native";

import FavoriteShops from "../FavoriteShops";
jest.mock("@react-navigation/native", () => {
  return {
    useIsFocused: () => true,
  };
});

describe("testing user favorite shops", () => {
  it("renders correctly", () => {
    const push = jest.fn();
    const tree = renderer.create(<FavoriteShops navigation={{ push }} />);
    expect(tree).toMatchSnapshot();
  });

  it("favorite shops fetch test", async () => {
    const fetchMock = require("fetch-mock-jest");
    const favorite_shop = [
      {
        title: "فروشگاه سر کوچه",
        address: "مرزداران",
        logo: "sare-kooche",
        rate_value: 2.3,
        online: true,
        phone: "09123456789",
        id: 0,
      },
      {
        title: "جانبو",
        address: "سعادت آباد",
        logo: null,
        rate_value: 3,
        online: false,
        phone: "09123456784",
        id: 1,
      },
      {
        title: "افق کوروش",
        address: "میدون شوش",
        logo: "shoosh",
        rate_value: 4.2,
        online: false,
        phone: "09156456789",
        id: 2,
      },
    ];

    var url = "http://eunoia-bshop.ir:8000/users/profile/likedshops";
    fetchMock.get(url, favorite_shop);

    const push = jest.fn();
    const { getByTestId, queryByTestId } = await render(
      <FavoriteShops navigation={{ push }} />
    );
    await new Promise((resolve) => setImmediate(resolve));

    expect(fetchMock).toHaveFetched(
      "http://eunoia-bshop.ir:8000/users/profile/likedshops",
      "get"
    );
    // console.log(getByTestId("favorite_shop-list").props);
    expect(getByTestId("fav-shops-list").props.data.length).toBe(
      favorite_shop.length
    );

    for (var i = 0; i < favorite_shop.length; i++) {
      const shop_name = favorite_shop[i].title.includes("فروشگاه")
        ? favorite_shop[i].title
        : "فروشگاه " + favorite_shop[i].title;
      var address = "آدرس: " + favorite_shop[i].address;

      //check shop name
      expect(getByTestId("shop-name-" + i).props.children).toBe(shop_name);

      //check shop address
      expect(getByTestId("shop-add-" + i).props.children).toBe(address);

      // // if item has image, show its own image
      // if (favorite_shop[i].photo) {
      //   var ph = "http://eunoia-bshop.ir:8000" + favorite_shop[i].photo;
      //   expect(queryByTestId("shop-image-" + i).props.source.uri).toBe(ph);
      //   // expect(queryByTestId("item-noimage-" + i)).toBeNull();
      // }
      // // else show default image
      //   else {
      //     console.log("this is image id", i);
      //     console.log(queryByTestId("item-noimage-" + i));
      //     // expect(queryByTestId("item-noimage-" + i)).not.toBeNull();
      //     // expect(queryByTestId("item-image-" + i)).toBeNull();
      //   }
    }

    fetchMock.mockClear();
  });
});
