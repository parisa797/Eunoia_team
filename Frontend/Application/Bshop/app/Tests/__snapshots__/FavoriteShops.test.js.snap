// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`testing user favorite shops renders correctly 1`] = `
<RCTScrollView
  nestedScrollEnabled={true}
  style={
    Object {
      "flex": 1,
      "marginBottom": "5%",
      "marginTop": "5%",
    }
  }
>
  <View />
</RCTScrollView>
`;
